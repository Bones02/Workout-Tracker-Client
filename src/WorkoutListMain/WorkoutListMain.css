@media only screen 
and (min-device-width: 300px) 
and (max-device-width: 500px) { 
    .WorkoutListMain {
        justify-content: center;
    }
    .workouts_header{
        text-align: center;
        margin-bottom: 0px;
    }
    .WorkoutListMain ul {
        list-style-type: none;
        margin: auto;
        padding: 10px;
        
    }
    .WorkoutListMain__button-container {
        margin: auto;  
        margin-bottom: 25px;
    }
    .WorkoutListMain__add-workout-button {
        margin: auto;
    }
    .WorkoutListMain__add-workout-button:hover {
        background-color: #29862e;
        color: white;
    }
    h2 {
        font-size: 14px;
    }
}

@media only screen 
and (min-device-width: 501px) {
    .workouts_header{
        text-align: center;
        margin-bottom: 20px;
    }

    .WorkoutListMain ul {
        list-style-type: none;
        margin: auto;
        padding: 10px;
    }
    
    .WorkoutListMain__button-container {
        margin: auto;
        margin-bottom: 25px;  
    }
    
    .WorkoutListMain__add-workout-button {
        margin: auto;
    }

    .WorkoutListMain__add-workout-button:hover {
        background-color: #29862e;
        color: white;
    }

    h2 {
        font-size: 32px;
    }
}
