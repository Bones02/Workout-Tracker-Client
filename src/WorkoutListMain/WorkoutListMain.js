import React from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Workout from '../Workout/Workout'
import Dashboard from '../Dashboard/Dashboard'
import CircleButton from '../CircleButton/CircleButton'
import ApiContext from '../ApiContext'
import { getWorkoutsForType} from '../workout-helpers'
//import './WorkoutListMain.css'

export default class WorkoutListMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

    handleDeleteNote = () => {
    this.props.history.push('/')
  };
  

  render() {
    const { workouts=[] } = this.context
    const typeId = this.props.match.params.typeId
    const workoutsForType = getWorkoutsForType(workouts, typeId)
    return (
      <section className='WorkoutListMain'>
          <Dashboard/>
        <ul>
          {workoutsForType.map(workout =>
            <li key={workout.id}>
              <Workout
                id={workout.id}
                name={workout.name}
                modified={workout.modified}
              />
            </li>
          )}
        </ul>
        <div className='WorkoutListMain__button-container'>
          <CircleButton
            tag={Link}
            to='/App/add-workout'
            type='button'
            className='WorkoutListMain__workout-button'>
            <FontAwesomeIcon icon='plus' />
            <br />
            Workout
          </CircleButton>
        </div>
      </section>
    )
  }
}
