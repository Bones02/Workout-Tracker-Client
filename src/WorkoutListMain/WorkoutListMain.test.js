import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import WorkoutListMain from './WorkoutListMain'

describe(`WorkoutListMain component`, () => {
  it('renders a .WorkoutListMain by default', () => {
    const wrapper = shallow(<WorkoutListMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
