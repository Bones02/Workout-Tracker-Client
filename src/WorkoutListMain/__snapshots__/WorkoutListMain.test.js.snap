// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutListMain component renders a .WorkoutListMain by default 1`] = `
<section
  className="WorkoutListMain"
>
  <Dashboard
    match={
      Object {
        "params": Object {},
      }
    }
  />
  <h2
    className="workouts_header"
  >
    Workouts
  </h2>
  <ul />
  <div
    className="WorkoutListMain__button-container"
  >
    <NavCircleButton
      className="WorkoutListMain__workout-button"
      tag={[Function]}
      to="/App/add-workout"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Workout
    </NavCircleButton>
  </div>
</section>
`;
